### ISSUE TITLE
Flag shortname directly preceding option

### ISSUE THREAD

**Issue by xealot** [alecthomas/kingpin#26]
It would be nice if shortname flags did not require a space

`cli -u username` <- valid

`cli -uusername` <- not valid

**Comment by alecthomas** [comment]
I kind of agree, but the underlying reason for this is that Kingpin currently separates lexing and parsing, and `-uusername` lexes as `-u -u -s -e -r -n -a -m -e` due to short-flag combining.

I guess the parser could detect a short flag with an argument and recombine subsequent short flags, but by that stage it has no way to differentiate between the inputs `-u -u` and `-uu`. So to make this really work I would need to change this behaviour.

**Comment by xealot** [comment]
I see. It would definitely have to understand more state within the system (to know that -u is a flag we're looking for). But then you can pretty easily get into these subtle errors where you have ambiguous parsing based on what options are available.

**Comment by alecthomas** [comment]
Yes, exactly. I erred on the side of correctness initially. But, as I've used Kingpin more and more (as a user) I do find the current flag parsing a bit annoying.

Another annoying aspect of the strictness is that, as a user, I would like to be able to specify any flags at any point, rather than strictly being positioned directly after the command they are associated with. The current behaviour is quite annoying for users.
